### ISSUE TITLE
UnpicklingError of the OAuth key

### ISSUE THREAD

**Issue by charlesg** [insanum/gcalcli#736]
This was working very well for months and started failing, sometime in the last 10 days I would think. I checked the GCP project and the credentials are still there, but I now get an UnpicklingErrors trying to run gcalcli. I'm dumping the stack trace below.
I was using the homebrew version, the test below is a freshly installed git clone. I'm getting the same error.

This is running on a MBP M2 Max running Sonoma 14.6.1.

```
Traceback (most recent call last):
  File "/Users/cgagnon/anaconda3/bin/gcalcli", line 8, in <module>
    sys.exit(main())
  File "/Users/cgagnon/anaconda3/lib/python3.10/site-packages/gcalcli/cli.py", line 151, in main
    gcal.AgendaQuery(start=parsed_args.start, end=parsed_args.end)
  File "/Users/cgagnon/anaconda3/lib/python3.10/site-packages/gcalcli/gcal.py", line 1190, in AgendaQuery
    return self._display_queried_events(start, end)
  File "/Users/cgagnon/anaconda3/lib/python3.10/site-packages/gcalcli/gcal.py", line 1134, in _display_queried_events
    event_list = self._search_for_events(start, end, search)
  File "/Users/cgagnon/anaconda3/lib/python3.10/site-packages/gcalcli/gcal.py", line 1096, in _search_for_events
    event_list.extend(
  File "/Users/cgagnon/anaconda3/lib/python3.10/site-packages/gcalcli/gcal.py", line 1041, in _GetAllEvents
    self.get_events()
  File "/Users/cgagnon/anaconda3/lib/python3.10/site-packages/gcalcli/gcal.py", line 183, in get_events
    return self.get_cal_service().events()
  File "/Users/cgagnon/anaconda3/lib/python3.10/site-packages/gcalcli/gcal.py", line 178, in get_cal_service
    credentials=self._google_auth())
  File "/Users/cgagnon/anaconda3/lib/python3.10/site-packages/gcalcli/gcal.py", line 135, in _google_auth
    self.credentials = pickle.load(gcalcli_oauth)
_pickle.UnpicklingError: invalid load key, '{'.
```

**Comment by dbarnett** [comment]
Good news is I think this was already addressed in #733, bad news is there's not a seamless fix and you'll probably need to redo authentication to get it working again (that change just provides a better error message to explain how to fix it).

**Comment by dbarnett** [comment]
Alright, so this wrinkle was still really bugging me so I hacked together a fallback so that people who upgrade and still have the old gcalcli_oauth file _should_ get it converted automatically without having to redo auth.
